lib/flutter_zoom_drawer.dart

```diff
@@ -380,108 +552,160 @@ class _ZoomDrawerState extends State<ZoomDrawer>
+        child: _mainScreen,
       );
     }
-    return _mainScreenContent;
+
+    // Add layer - AbsorbPointer
+    /// Prevents touches to mainScreen while drawer is open
+    if (widget.mainScreenAbsorbPointer) {
+      _mainScreen = Stack(
+        children: [
+          _mainScreen,
+          ValueListenableBuilder(
+            valueListenable: _absorbingMainScreen,
+            builder: (_, bool _valueNotifier, ___) {
+              if (_valueNotifier && stateNotifier.value == DrawerState.open) {
+                return AbsorbPointer(
+                  child: Container(
+                    color: Colors.transparent,
+                    width: MediaQuery.of(context).size.width,
+                    height: MediaQuery.of(context).size.height,
+                  ),
+                );
+              }
+              return const SizedBox.shrink();
+            },
+          ),
+        ],
+      );
+    }
+
+    return _mainScreen;
   }
 
   @override
   Widget build(BuildContext context) {
-    if (widget.disableGesture) return renderLayout();
-
     return GestureDetector(
-      /// Detecting the slide amount to close the drawer in RTL & LTR
-      onPanUpdate: (details) {
-        if (_state == DrawerState.open &&
-            ((details.delta.dx < -widget.swipeOffset && !widget.isRtl) ||
-                (details.delta.dx > widget.swipeOffset && widget.isRtl))) {
-          toggle();
-        }
-      },
-      onTap: () {
-        if (widget.mainScreenTapClose && _state == DrawerState.open) {
-          return close();
-        }
-      },
+      onTap: _onTap,
       child: renderLayout(),
     );
   }
 
   Widget renderLayout() {
-    if (widget.drawerStyleBuilder != null) return renderCustomStyle();
+    Widget _parentWidget = renderDefault();
+
     switch (widget.style) {
       case DrawerStyle.style1:
-        return renderStyle1();
+        _parentWidget = renderStyle1();
+        break;
       case DrawerStyle.style2:
-        return renderStyle2();
+        _parentWidget = renderStyle2();
+        break;
       case DrawerStyle.style3:
-        return renderStyle3();
+        _parentWidget = renderStyle3();
+        break;
       case DrawerStyle.style4:
-        return renderStyle4();
+        _parentWidget = renderStyle4();
+        break;
       case DrawerStyle.style5:
-        return renderStyle5();
+        _parentWidget = renderStyle5();
+        break;
       case DrawerStyle.style6:
-        return renderStyle6();
+        _parentWidget = renderStyle6();
+        break;
       case DrawerStyle.style7:
-        return renderStyle7();
+        _parentWidget = renderStyle7();
+        break;
       case DrawerStyle.style8:
-        return renderStyle8();
+        _parentWidget = renderStyle8();
+        break;
       default:
-        return renderDefault();
+        _parentWidget = renderDefault();
     }
+
+    if (widget.drawerStyleBuilder != null) _parentWidget = renderCustomStyle();
+
+    return WillPopScope(
```

Comment:
Since this is specific to Android, add an `if` check and only wrap the content with `WillPopScope` when in Android

Reply:
In order to use `Platform.isAndroid` we need to import `import 'dart:io';`

Which will breaks the package for building to Web.

Reply:
You can check fie kisweb first and than check for isandroid. With this it won't break